rand/insecure.go

```diff
@@ -0,0 +1,82 @@
+package rand
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890"
+
+var (
+	// This is the formula for the number of bits that is needed to represent
+	// the number of characters in the letters
+	indexBits = uint(math.Floor(math.Log2(float64(len(letters)))) + 1)
+
+	// The mask to use to select random bits
+	indexMask = int64(1)<<indexBits - 1
+
+	// The number of indices we can read using 63 bits
+	numReads = 63 / indexBits
+
+	// A custom source for non-concurrent InsecureRand calls
+	src = rand.NewSource(time.Now().UnixNano())
+)
+
+// This optimized version is heavily based off of this wonderful Stack Overflow
+// answer:
+// https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-go
+
+// InsecureRand returns a random string of the specified length that is
+// sufficiently random for use in identifiers but should not ever be used to
+// seed cryptographic operations. This should not be used concurrently
+func InsecureRand(n int) string {
```

Comment:
Remove the `rand` suffix since it's in the rand package maybe? `InsecureSafe` also.

```suggestion
func Insecure(n int) string {
```

Reply:
Yeah, that makes sense. Should it be `InsecureString` though?

Reply:
Would it be crazy to drop the `Insecure` prefix and just go with `String` and `StringSafe`? We have a disclaimer in the godoc anyway.

Also— what do you think about flipping the naming convention so that we "default" to safe? i.e. `String` and `StringUnsafe`?

Reply:
That isn't a bad idea on the defaulting. However, I was thinking we'd put the secure rand in here as well. Should we put that in a `crypto/rand` package instead?

Reply:
`StringSecure` maybe?

Reply:
So `String`, `StringUnsafe`, `StringSecure`?

Reply:
yessir — that's what I'm thinking